test/test_path_tracking_pid_fw_bw.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+import unittest
+from math import hypot
+
+import rospy
+import rostest
+from actionlib import GoalStatus as GS
+from actionlib import SimpleActionClient
+from dynamic_reconfigure.client import Client as ReconfigureClient
+from mbf_msgs.msg import ExePathAction, ExePathGoal
+from nav_msgs.msg import Odometry
+
+from paths import create_path
+
+
+class TestPathTrackingPID(unittest.TestCase):
+    def setUp(self):
+        self.cur_odom = Odometry()
+        self.prev_odom = Odometry()
+        self.cur_accel = 0.0
+
+    def reconfigure(self, target_vel):
+        reconfigure = ReconfigureClient("/move_base_flex/PathTrackingPID", timeout=5)
+        reconfigure.update_configuration({"target_x_vel": target_vel})
+        reconfigure.update_configuration({"target_end_x_vel": 0})
+        reconfigure.update_configuration({"target_x_acc": 4.0})
+        reconfigure.update_configuration({"target_x_decc": 1.0})
+        reconfigure.update_configuration({"use_mpc": False})
+
+    def odom_cb(self, msg):
+        self.prev_odom = self.cur_odom
+        self.cur_odom = msg
+        dv = self.cur_odom.twist.twist.linear.x - self.prev_odom.twist.twist.linear.x
+        dt = (self.cur_odom.header.stamp - self.prev_odom.header.stamp).to_sec()
+        self.cur_accel = dv / dt
+
+    def test_exepath_action(self):
+
+        # Setup action client to ask question
+        client = SimpleActionClient("move_base_flex/exe_path", ExePathAction)
+        self.assertTrue(
+            client.wait_for_server(timeout=rospy.Duration(10)),
+            msg="No actionclient for recipe_control found",
+        )
+
+        # Subscribe to pose messages from
+        self.pose_sub = rospy.Subscriber("/odom", Odometry, self.odom_cb)
+
+        path = create_path([0.0, 10.0], [0.0, 0.0], [0.0, 0.0])
+
+        # Start goal and evaluate outcome
+        outcome_exp = rospy.get_param("~outcome", GS.SUCCEEDED)
+        self.reconfigure(2.0)
+        rospy.logwarn("Starting path!")
+        client.send_goal(ExePathGoal(path=path))
+
+        rospy.sleep(0.5)
+        self.assertTrue(3.5 < self.cur_accel < 4.5, msg="Violated acceleration {}".format(self.cur_accel))
+        rospy.sleep(4.5)
```

Comment:
I'm a bit worried about these sleeps. Are you using `use_sim_time` during testing? If not these sleeps can take longer due if there is more load on the runner

Reply:
Nope we're not, but we're using `mobile_robot_simulator` here which simply runs on ROS-clock (opposed to i.e. gazebo who publishes the ROS-clock).

Reply:
We do have issues with these things @Rayman. Therefor the margin should be bigger than desired, in case CI is slow.

But since this is in line with the other tests I'm good here.